cmd/log_mgmt/src/log_mgmt.c

```diff
@@ -310,6 +313,11 @@ log_encode_entries(const struct log_mgmt_log *log, CborEncoder *enc,
         return LOG_MGMT_ERR_ENOMEM;
     }
 
+#if LOG_MGMT_READ_WATERMARK_UPDATE
+    if (!rc) {
+        rc = log_mgmt_impl_set_watermark(log, ctxt.last_enc_index);
```

Comment:
You probably shouldn't assign `rc` here.  By assigning `rc`, we will send an error response if we fail to set the watermark.  One reason this function might fail is that the log doesn't support watermarks.  We don't want to make those logs unreadable!

Reply:
Thanks Chris, I’ll take care of this, good catch!